---
title: "MICB425 portfolio"
author: "Polina Beskrovnaya (38929148)"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document:
    toc: true
 
---
# Module 01 - Origins and Earth Systems


## Data science Friday
### Installation check
Space for installation screenshots:
![Installation screenshot 1](1.jpg)
![Installation screenshot 2](2.jpg)


### Portfolio repo setup
**code you used to create, initialize, and push your portfolio repo to GitHub:**

cd ~/Documents/MICB425_portfolio

git init

git add .

git commit -m "First commit"

git remote add origin https://remote_repository_URL

git remote -v

git push -u origin master

### RMarkdown pretty PDF challenge
**Paste your code from the in-class activity of recreating the example PDF:**

This is my attempt at creating a pretty document in R Studio. I am sort of trying, but coding is very confusing. 

The following assignment is an exercise for the reproduction of this .html document using the RStudio and RMarkdown tools we?ve shown you in class. Hopefully by the end of this, you won?t feel at all the way this poor PhD student does. We?re here to help, and when it comes to R, the internet is a really valuable resource. This open-source program has all kinds of tutorials online.

*Here is a picture ~~I was forced to add~~ to distract you from my feeble attempts at coding:*

![**Some comic.**](http://phdcomics.com/comics/archive/phd011718s.gif){width=70%}

*And here is a comic I like a little bit more:*

![**A slightly more relatable comic...**](http://phdcomics.com/comics/archive/phd101609s.gif){width=70%}

**Both comics were taken from** [**PhD Comics.**](http://phdcomics.com)

####Challenge Goals

The goal of this R Markdown html challenge is to give you an opportunity to play with a bunch of different RMarkdown formatting. Consider it a chance to flex your RMarkdown muscles. Your goal is to write your own RMarkdown that rebuilds this html document as close to the original as possible. So, yes, this means you get to copy my irreverant tone exactly in your own Markdowns. It?s a little window into my psyche. Enjoy =)

**hint: go to the [PhD Comics website](http://phdcomics.com) to see if you can find the image above**
*If you can?t find that exact image, just find a comparable image from the PhD Comics website and include it in your markdown*

#####I am not sure I am enjoying this assignment yet.

And my feelings deserve a whole separate section. You may even have noticed that this is under a **different** level header.

And since we are sharing comics, I want to make room for one more:

![[Explosm.net](http://explosm.net)](http://files.explosm.net/comics/Rob/housefiah.png){width=70%}

####Here's a header! Mathy header. 

Let?s be honest, this header is a little arbitrary. But show me that you can reproduce headers with different levels please. This is a level 3 header, for your reference (you can most easily tell this from the table of contents).

#####Another header, now with maths

Perhaps you?re already really confused by the whole markdown thing. Maybe you?re so confused that you?ve forgotton how to add. Never fear! ~~A calculator~~ R is here:

I'm not sure why we are doing math, but:

```{r}
666+13131313131313134
```

**Wow, this worked!** I am shocked it^actually^ worked...

####Why I am taking this class:

```{r}
dat<- data.frame(t=seq(0, 2*pi, by=0.1) )
 xhrt <- function(t) 16*sin(t)^3
 yhrt <- function(t) 13*cos(t)-5*cos(2*t)-2*cos(3*t)-cos(4*t)
 dat$y=yhrt(dat$t)
 dat$x=xhrt(dat$t)
 with(dat, plot(x,y, type="l", xlab="Me", ylab="Bacteria"))
```

Is it bad that I took pretty codes from [somewhere?](https://stackoverflow.com/questions/8082429/plot-a-heart-in-r)

####Here is a table:

Or maybe, after you?ve added those numbers, you feel like it?s about time for a table!
I?m going to leave all the guts of the coding here so you can see how libraries (R packages) are loaded into R (more on that later). It?s not terribly pretty, but it hints at how R works and how you will use it in the future. The summary function used below is a nice data exploration function that you may use in the ^future^.
```{r}
library(knitr)
kable(summary(cars),caption="I made this table with kable in the knitr package library, but I'm not sure why")
```

####Here's another table:

Organism   | Level of coolness(%)
---------  |------------------
Human      | 0
Giraffe    | 75
Polar bear | 100
Prokaryote | 100


####The big finale.

This gif sums up me vs. all the assignment deadlines and grad school application process this semester:

![](https://media.giphy.com/media/ZQ6YRgFGqZ9xC/giphy.gif)


###ggplot exercises
Code pasted here:

```{r}
library(tidyverse)
library(phyloseq)
```

```{r}
read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings="NAN")
metadata = read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t")
load("phyloseq_object.RData")

#The base function of any plot from ggplot is unsurprisingly the ggplot function. 
#This function takes in the data and basic aesthetics of the plot, 
#like the x- and y-axis variables, but does not specify the way 
#in which you want to plot the data (like dots, bars, etc.). 
#So, if we try to plot depth by oxygen (O2) with just ggplot, 
#we get a plot with the correct axes but no actual data.
ggplot(metadata, aes(x=O2_uM, y=Depth_m))

#To add in the data, we must specify a geom. To get a dot plot, we will use geom_point
ggplot(metadata, aes(x=O2_uM, y=Depth_m)) +
  geom_point()

#So, the above is equivalent to
ggplot(metadata) +
  geom_point(aes(x=O2_uM, y=Depth_m))

#We can change the color (outline and fill), shape, or size of our points
ggplot(metadata, aes(x=O2_uM, y=Depth_m, color="blue")) +
  geom_point()

#But wait! That’s not blue! What happened?
#Within ggplot, you can specify attributes that are part of the overall plot 
#(everything that shows up if you only use ggplot, no geom) 
#but not attributes associated with a specific geom. 
#In this case, the dots are there because of geom_point 
#so we must change their color within the geom function.
ggplot(metadata, aes(x=O2_uM, y=Depth_m)) +
  geom_point(color="blue")

#The same holds for changing shape and size such as in
ggplot(metadata, aes(x=O2_uM, y=Depth_m)) +
  geom_point(shape="square")
ggplot(metadata, aes(x=O2_uM, y=Depth_m)) +
  geom_point(size=10)

#Let’s see if the oxygen measurement in volts correlates with measurements in μμM. 
#In this case, we will need a legend, which is part of the base ggplot 
#so we must include size within the geom instead of the base ggplot. 
ggplot(metadata, aes(x=O2_uM, y=Depth_m, size=OxygenSBE_V)) +
  geom_point()

#Exercise 1
#Plot another nutrient of your choice against depth. 
#Change the points to be purple triangles.
ggplot(metadata, aes(x=Fluorescence_mgm_3, y=Depth_m)) +
  geom_point(color="purple", shape=17,size=3)

#Exercise 2
#Using dplyr, convert the temperature variable from Celsius to Fahrenheit. 
#Then create a dot plot of temperature in Fahrenheit against depth.
#Step 1: find name of column with temperature values
metadata %>% 
  select(matches("Temp"))
#Step 2: mutate values celsius->farenheit and show graph
#My attempt
metadata %>%
  mutate(Temperature_F=(Temperature_C*1.8)+32) %>%
  ggplot(aes(x=Temperature_F, y=Depth_m)) +
  geom_point()

#I first couldn't figure out how to save mutated variable so it could be used for plot...
#So I made the plot like this:
ggplot(metadata, aes(x=Temperature_C*1.8+32, y=Depth_m)) +
  geom_point() +
  xlab("Temperature_F")

#ggplot with phyloseq
physeq
#phyloseq plot functions mirror ggplot like plot_bar = geom_bar. 
#For example, we can plot the phylum level communities by depth 
#by simply specifying the fill.
plot_bar(physeq, fill="Phylum")
#we can calculate a percentage of the total of each sample for each OTU 
#with the following. The function applies the equation we give it 
#(100*x/sum(x)) to every cell in our OTU table
physeq_percent = transform_sample_counts(physeq, function(x) 100 * x/sum(x))
plot_bar(physeq_percent, fill="Phylum")
#To get rid of black lines:
plot_bar(physeq_percent, fill="Phylum") + 
  geom_bar(aes(fill=Phylum), stat="identity")

#Exercise 3
#Create a bar plot at a different taxonomic level 
#with more descriptive x- and y-axis labels as well as a title. 
plot_bar(physeq_percent, fill="Order") + 
  geom_bar(aes(fill=Order), stat="identity") +
  ggtitle("Orders from 10 to 200 m in the Saanich Inlet") +
  ylab("Percent relative abundance") + xlab("Sample depth")

#Faceting
#some low abundance taxa are so low that you can’t see them 
#and some colors are so similar that it’s difficult to tell them apart.
#separate it into multiple panels (or facets) to more clearly see patterns. 
#Working from our previous bar plot, we simply add face_wrap 
#and tell it what variable to separate by
plot_bar(physeq_percent, fill="Phylum") +
  geom_bar(aes(fill=Phylum), stat="identity") +
  facet_wrap(~Phylum)
#We can let R scale independently in each panel with the scales parameter like so.
#We also don’t really need a legend anymore because each facet it labeled. 
#So we can remove this in the theme.
plot_bar(physeq_percent, fill="Phylum") +
  geom_bar(aes(fill=Phylum), stat="identity") +
  facet_wrap(~Phylum, scales="free_y") +
  theme(legend.position="none")

#Exercise 4
#Using ggplot, create a faceted figure showing nutrient concentrations 
#in uM for O2, PO4, SiO2, NO3, NH4, and NO2 by depth. 
#For this, you can use either the metadata table or the phyloseq object.
gather(metadata, key="nutrients", value="uM", O2_uM, PO4_uM, SiO2_uM, NO3_uM,NH4_uM, NO2_uM) %>%
ggplot(aes(x=Depth_m, y=uM)) +
  geom_path() + facet_grid(~nutrients)

```

## Evidence worksheets

###Evidence worksheet 01 
The template for the first Evidence Worksheet has been included here. The first thing for any assignment should link(s) to any relevant literature (which should be included as full citations in a module references section below).

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems. 

#### General questions
* What were the main questions being asked?
1. What is the abundane of prokaryotes in different environments & their total abundance?
2. How does C,N,P content of prokaryotes relate to that of plants, animals, etc.?
3. What is the potential for prokaryotic genetic diversity?
4. What are the different approaches for estimating abundances of prokaryotes? How do they compare? What are their limitations?

* What were the primary methodological approaches used? 
Cell counts from other studies + estimates based on volumes, area, etc. Extrapolation.

* Summarize the main results or findings.  
1. C pool of prokaryotes is almost as large as in plants.
2. N,P more than in plants.
3. Prokaryotes some of most abundant types of organisms in general + most abundant in some environments.
4. Prokaryotes account for most 1' productivity in oceans.
5. Abundance + mutation rates = potential for large unexplored genetic diversity.

* Do new questions arise from the results?  
1. How can we explore genetic diversity?
2. How do we revise the definition of prokaryote species?
3. Are estimates accurate? How can we check them?
4. Are there biases, limitations, improvements possible?
5. How do prokaryote numbers inside living organisms actually compare to non-living environments?
6. Are there new estimates? Are they different?

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
1. What is the significance of turnover rates?
2. Species definition discussion was unexpected.
3. No formulas provided.
4. Not clear how exactly estimates were calculated.

### Evidence worksheet 02

[Nisbet & Sleep 2001](https://www.nature.com/articles/35059210)

[Kasting & Siefert 2003](https://www.ncbi.nlm.nih.gov/pubmed/12004117)

#### Learning objectives: 
Comment on the emergence of microbial life and the evolution of Earth systems

####Questions:
* Indicate the key events in the evolution of Earth systems at each approximate moment in the time series. If times need to be adjusted or added to the timeline to fully account for the development of Earth systems, please do so. 

    + 4.6 billion years ago  
    Formation of Earth
    
    + 4.1 billion years ago  
    Oldest record of life on Earth (fixed C in solid material)
    
    + 3.8 billion years ago 
    Though to be when first life actually appeared (single-celled prokaryotes)
    
    + 3.75 billion years ago
    Evidence of banded iron formations
    
    + 3.5 billion years ago 
    Oldest fossil evidence of life in rock, measured by radiometric dating - microfossils. Photosynthesis emerged (oxygenic)
    
    + 3.0 billion years ago  
    First continent formed
    
    + 2.7 billion years ago  
    Cyanobacteria form oxygen-rich atmosphere
    
    + 2.2 billion years ago  
    End of Great Oxidation Event
    
    + 2.1 billion years ago
    Multicellularity emerges
    
    + 1.3 billion years ago
    First plants appear
    
    + 550,000 years ago
    Multicellular animals appear
    
    + 200,000 years ago
    Mass extinction


* Describe the dominant physical and chemical characteristics of Earth systems at the following waypoints:  

    + Hadean  
    High temperatures, sometimes enough to turn oceans to steam. Formation of atmosphere around Earth, depleted in oxygen but rich in CO2, methane, ammonia. Frequent asteroid bombardments. Chemistry dominated by volcanic activity -- "primeval soup".Sun was also dimmer and cooler
    
    + Archean  
    Carbon fixed in rock as evidence of emergence of oxygenic photosynthesis. Oxygen concentrations rise in the atmosphere, but still not much. Formation of solid crust due to cooling of temperatures
    
    + Precambrian  
    Atmosphere first composed of inert gases, CO2, ammonia, then enriched in oxygen once photosynthesis appears. Formation of crust, plate tectonics, continents. Banded iron formations appear
    
    + Proterozoic
    Oxygen increasing in atmosphere, frequent glaciations. "Snowball Earth". Supercontinents and emergence of multicellular life.
    
    + Phanerozoic  
    Tectonic activity and separation of supercontinents. Rapid evolution rates. Atmosphere similar to current and conditions overall as well are similar. Mass extinction events.
    
### Evidence worksheet 03

[RockstrÖm 2009](https://www.nature.com/articles/461472a)

#### Learning objectives: 
Evaluate human impacts on the ecology and biogeochemistry of Earth systems

####Questions:
* What were the main questions being asked?

1. How do we define the current geologic era?
2. What are its features?
3. How do we preserve humanity through it?
4. What are the associated thersholds?

* What were the primary methodological approaches used?

Proposal of planetary boundaries: define critical values

1. atmospheric CO2, radiative forcing
2. rate of biodiversity loss by extinction statistics
3. N, P cycles by turnover


* Summarize the main results or findings. 

Established critical values for climate change, biodiversity loss, cycles of N and P. If humanity maintains the Earth within the established boundaries, it can survive

* Do new questions arise from the results?

1. Are the estimates of critical values reliable?
2. How can we check?
3. Were some sources of uncertainty omitted?

* Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

It was hard to understand general concept of critical values/thresholds


## Problem sets

###Problem set 01

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:
* What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  

1. Aquatic habitats: 1180 * 10^26 cells total
  low oxygen diffusion and light penetration, low nutrient concentrations unless close to shore or surface
2. Subsurface sediments: 380 * 10^28 cells total
  low availability of favourable energy sources, electron acceptors, other nutrients; no light or oxygen
3. Soil: 255.6 * 10^27 cells total
  more nutrients and oxygen, esp when supplied by decomposing or living higher organisms; moisture is crucial
4. Animals: negligible compared to total
  varies a lot based on host, lots of competition
  

* What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  

360 * 10^26 cells, of which most are autotrophs (10^28) such as Prochlorococcus. Most C supplied by these organisms to allow heterotrophic bacteria to survive. Also oceanic cyanobacteria contribute ~50% of world's atmospheric O2

* What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?  

How much each relies on supply of organic C for energy and how much is produced by each type. Autotrophs generate C, heterotrophs consume it. Lithotrops do not strictly consume organic C, but have additional energy sources they could utilize

* Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  

Availability of C for assimilation since 85% of primary productivity consumed in the upper layers. assume 4 km for max depth

* Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?  

77 km is the highest altitude where prokaryotes have been detected, but availability of C for assimilation is also a limiting factor (as well as other nutrients)

* Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth’s biosphere measured in km?  

4 + 77 km = 81 km

* How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)  

numbers of cells by extrapolation (e.g. 3*10^23) x estimated turnover rate (e.g. 16 days for whole population) = cellular production


* What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  

C content in cell generated by assimilation and recycled (turnover) when cell dies and is decomposed. Rates depend on availability of light, oxygen, other nutrients available to organisms consuming C for growth. Lots of mixing in the ocean (dispersion), but C produced by autotrophs sinks down and is quickly consumed generating gradient. Terrestrial environments not uniform and different because of that.

* How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  

((Mutations/replication)x(replications/cell)x(cells/year))^4

* Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt?  

Diversity can also arise through gene losses, suplications, inversions, HGT, etc. Since lots of cells in diverse environments = diverse genomes. High turnover rates for all cells in total --> lots of mutations possible.

* What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text? 

Prokaryotes have undeniably large potential for genetic diversity, protein diversity, etc. Given that there are many envornment types = many adaptations required and made possible, coupled with turnover rates and large abundances.

### Problem set 02

[Falkowski *et al.* 2008](https://www.ncbi.nlm.nih.gov/pubmed/18497287)

#### Learning objectives: 
Discuss the role of microbial diversity and formation of coupled metabolism in driving global biogeochemical cycles.

#### Specific Questions:
* What are the primary geophysical and biogeochemical processes that create and sustain conditions for life on Earth? How do abiotic versus biotic processes vary with respect to matter and energy transformation and how are they interconnected?  

1. Geophysical: tectonics, atmospheric photochemical processes.
    + Mostly based on acid-base chemistry
    + Bond form and break in cyclical manner
    + Resupply C, S, P
    
2. Biogeochemical: redox reactions involving major elements (C, H, O, N, S, P)
    + Relatively limited set of chemical energy
    + Coupled half-cells linked to elemental cycles
    + Set lower limits on external energy

* Why is Earth’s redox state considered an emergent property?

    + biogeochemical rxns driven abiotically, rely on supply of elements from geophysical, but due to this allow for setting lower limites on external energy required to sustain elemental cycles --> stabilize Earth's redox state
    + biological oxidation driven by photosynthesis --> feed cycles of elements that are maintained by forward-reverse redox
  

* How do reversible electron transfer reactions give rise to element and nutrient cycles at different ecological scales? What strategies do microbes use to overcome thermodynamic barriers to reversible electron flow?  

Close association between organisms feeding the cycle, e.g. waste products of 1' producer used for energy by another organisms that produce waste, which serves as substrate for next --> remineralization.
Shifting availability of substrates/products to make forward/reverse rxns thermodynamically favourable.

* Using information provided in the text, describe how the nitrogen cycle partitions between different redox “niches” and microbial groups. Is there a relationship between the nitrogen cycle and climate change?  

Conserved machinery in diverse taxa, specialized groups
N fixation is diverse, but nitrification by bacteria and archaea --> anaerobic: oxidize organic matter into CO2, greenhouse gas released into atmosphere

* What is the relationship between microbial diversity and metabolic diversity and how does this relate to the discovery of new protein families from microbial community genomes?  

Subset of core conserved genes + environment-specific genes. Large unexplored sequence space creates illusion of limitless diversity, but new proteins aren't discovered as frequently. Functions and adaptations limited by protein-protein or protein-nucleic acid interactions and biochemistry in general.

* On what basis do the authors consider microbes the guardians of metabolism?  

Functional units of the pathway can go extinct, but the process survives. Some organisms under the right conditions store/utilize pathway or its steps with a chance to pass it on to other organisms by HGT

##Writing assignment 1

* Prompt: Microbial life can easily live without us; we, however, cannot survive without the global catalysis and environmental transformations it provides.

**Introduction**

In the world undergoing rapid changes due to increasing rates of perturbations from human activity, it is becoming clear that we are not prepared to face the consequences of our actions. Microbes, on the other hand, are ubiquitous and functionally versatile, which allows them to quickly adapt to even the most dramatic changes to the environment. With this, it only seems logical that we will ultimately be outlived by microorganisms, and that without them our own survival would be under question. 

**Genomic diversity of microbes mediates functional versatility**

Microbial life has long been able to sustain itself, with its history possibly dating as far back as 4.1 billion years ago – as recorded in geological evidence – in contrast to more complex multicellular organisms, which are only thought to have emerged approximately 600-800 million years ago (1, 2). Time has allowed prokaryotes to develop exquisite varieties of biological machinery suitable for pretty much all types of environments existing on Earth, making them dominating lifeforms in even some of the most extreme places where no eukaryote could possibly survive. In other words, it is not just the hostile and methane-rich Early Earth, scarred by the constant volcanic activity, that would come to be monopolized by microbial life (3). 

For example, hypoxic areas of the world’s oceans, aptly named “dead zones” for their inability to support animal life, are, in fact, home to diverse bacterial communities, where taxa richness is often tied to decreasing oxygen concentrations (4, 5). Similarly, microbes are known to thrive in subsurface environments, also characterized by limited availability of oxygen, but typically providing resources for maintaining organismal function through processes such as metal ion reduction, methanogenesis and denitrification (6, 7). While these processes are less energetically efficient than habitual aerobic respiration, they are still largely responsible for the total estimated abundance of roughly 3.8×1030 cells in all subsurface environments (8). 

The rich metabolic diversity, coupled with the possibility of its conservation through millennia with the help of lateral gene transfer, makes microbes the only lifeforms suitable for independent functioning even under dramatic changes to the environmental conditions (3, 9, 10).

**Irreplaceable symbiosis traversing Domains of Life**

Human population, on the other hand, is not yet able to secure its own survival in a world devoid of microscopic organisms. Our dependence on microbially catalyzed processes is evident on all scales of life – from individual cells exploiting the ancient mitochondrial machinery for harvesting energy, to whole industries based on bacterial production of polymeric substances, foods and therapeutics, as well as bioremediation. 

At the organismal level, an average person harbours approximately 3.8×1030 bacterial cells, representing hundreds of species, all of which serve a particular purpose in the human body (11). Numerous members of the gut microbiome metabolize dietary substances immune to human enzymes, providing us with short-chain fatty acids, simple sugars and other important energy sources (12, 13). In addition, commensal microbes prove invaluable in maturation of the immune system, protecting us, their hosts, from viral and parasitic infections (14-16).

At the other end of the spectrum, we rely on microbial processes for large-scale production of antibiotics, food and beverages, and various industrial goods. While we often succeed in devising complex, multi-step pathways for synthesis or degradation of various chemicals, in practice, many of these transformations can only be performed with the help of a living microscopic cell – for example, production of vitamin B12 and insulin, as well as breakdown of recalcitrant polychlorinated biphenyl pollutants (17-19).

**Technological advancement at the cost of adaptability**

While we are finally becoming increasingly aware of the various roles that microscopic organisms play in supporting the Earth’s biosphere, we are still far from fully disentangling the interconnections between microbial chemistry, abiotic processes, functions of higher organisms, and our own place in this complex web of interactions. Even in places where complex organisms are plenty, microbes serve important ecological roles, rendering their eradication detrimental to not just the human population, but all remaining life on Earth. 

It has already been established that microbes largely contribute to primary production in the oceans, with the annual rates of carbon fixation estimated at 45 Pg C, which accounts for about half of total primary production on Earth – needless to say, this organic carbon fuels the food chain, which we all form a part of (9, 20, 21). Further, microbial contributions to the global nutrient cycles are simply too great to ignore. The oxygen-rich environment, which we live in today, was shaped by activity of cyanobacteria and other photosynthesizing microbes, all of which still account for the majority of total oxygen produced per year on Earth (3, 21). Similarly, recycling of nitrogen is largely carried out by prokaryotes, which are the only lifeforms on Earth capable of mediating turnover of N2 gas into its organic forms, which are more readily available to all other Domains of Life (9, 22). 

Although humans are attempting to catch up with their increasing ammonium demands through industrial nitrogen fixation, also known as the Haber-Bosch process, this transformation brings yet more caveats, such as massive rates of fossil fuel consumption, greenhouse gas emissions, and local nitrogen overaccumulation in the aquatic environments, which leads to toxic algal blooms (22-23). Together, these examples present one of the numerous illustrations of our own inability to foresee the consequences to our actions within the complex ecosystem. However, for someone who is lacking the genotypic diversity comparable to that of microorganisms, it is only necessary to always be able to think two steps ahead, if we are to ensure our own survival in the ever-changing world.

**Conclusion**

While microbes, due to their exquisite genotypic richness, are likely to quickly adapt and re-program interactions within their complex networks that would allow them to thrive on the new Earth, characterized by anthropogenic perturbations, we do not yet possess the technology nor the insight to assess full implications of our actions and their long-term effects on the biosphere (22, 24). The latter, coupled with the long history of yet irreplaceable human-microbe symbiosis, would render humanity as we know it today completely helpless in a world devoid of microorganisms. In order to preserve the humanity through the Anthropocene, we must find comfort in our newly rediscovered modesty and, like so many times before, turn to the microscopic world for inspiration. 

**References**

1.	Bell EA, Boehnke P, Harrison TM, Mao WL. 2015. Potentially biogenic carbon preserved in a 4.1 billion-year-old zircon. Proc Natl Acad Sci. 112:14518-14521.

2.	 Vavour T, Semple JI, Lehner B. 2008. Widespread conservation of genetic redundancy during a billion years of eukaryotic evolution. Trends Genet. 24:485-488. 

3.	Nisbet EG, Sleep NH. 2001. The habitat and nature of early life. Nature. 409:1083-1091.  

4.	Spietz RL, Williams CM, Rocap G, Horner-Devine MC. 2015. A Dissolved Oxygen Threshold for Shifts in Bacterial Community Structure in a Seasonally Hypoxic Estuary. PLoS One. 10:e0135731.  

5.	Beman JM, Carola MT. 2013. Deoxygenation alters bacterial diversity and community composition in the ocean's largest oxygen minimum zone. Nat Commun. 4:2705 

6.	Gulecal-Pektas Y, Temel M. 2016. A Window to the Subsurface: Microbial Diversity in Hot Springs of a Sulfidic Cave (Kaklik, Turkey). Geomicrobiol J. 34:374-384.

7.	Lovley DR, Chapelle FH. 1995. Deep subsurface microbial processes. Rev Geophys 33:365-381.  

8.	Whitma WB, Coleman DC, Wiebe WJ. 1998. Prokaryotes: the unseen majority. Proc Natl Acad Sci. 95:6578-6583  

9.	Falkowski PG, Fenchel T, Delong EF. 2008. The microbial engines that drive Earth's biogeochemical cycles. Science. 320:1034-1039. 

10.	Kasting JF, Siefert JL. 2002. Life and the Evolution of Earth's Atmosphere. Science. 296:1066-1068.  

11.	Sender R, Fuchs S, Milo R. 2016. Revised Estimates for the Number of Human and Bacteria Cells in the Body. PLoS Biol. 14:e1002533.

12.	 Shreiner AB, Kao JY, Young VB. 2015. The gut microbiome in health and in disease. Curr Opin Gastroenterol. 31:69-75.   

13.	 St Leger AJ, Desai JV, Drummond RA, Kugadas A, Almaghrabi F, Silver P, Raychaudhuri K, Gadjeva M, Iwakura Y, LionakisMS, Caspi RR. 2017. An Ocular Commensal Protects against Corneal Infection by Driving an Interleukin-17 Response from Mucosal γδ T Cells. Immunity. 47:148-158.  

14.	Nicholson JK, Holmes E, Kinross J, Burcelin R, Gibson G, Jia W, Pettersson S. 2012. Host-gut microbiota metabolic interactions. Science. 336:1262-7
15.	Robinson CM, Pfeiffer JK. 2014. Viruses and the Microbiota. Ann Rev Virol. 1:55-69.

16.	Zaiss MM, Harris NL. 2016. Interactions between the intestinal microbiome and helminth parasites. Parasite Immunol. 38:5-11. 

17.	Fang H, Kang J, Zhang D. 2017. Microbial production of vitamin B12: a review and future perspectives. Microb Cell Fact. 16:15.

18.	 Baeshen NA, Baeshen MN, Sheikh A, Bora RS, Ahmed MMM, Ramadan HAI, Saini KS, Redwan EM. 2014. Cell factories for insulin production. Microb Cell Fact. 13:141.

19.	Murínová S, Dercová K, Dudášová H. 2014. Degradation of polychlorinated biphenyls (PCBs) by four bacterial isolates obtained from the PCB-contaminated soil and PCB-contaminated sediment. Int Biodeterior Biodegrad. 91:52-59.

20.	Kirchman DL. 2011. Microbial primary production and phototrophy. Processes in microbial ecology. Oxford University Press, New York.

21.	Thornton DCO. 2012. Primary Production in the Ocean. Advances in Photosynthesis - Fundamental Aspects. InTech, London.

22.	Canfield DE, Glazer AN, Falkowski PG. 2010. The Evolution and Future of Earth's Nitrogen Cycle. Science. 330:192-196.

23.	Glibert PM, Maranger R, Sobota DJ, Bouwman L. 2014. The Haber Bosch-harmful algal bloom (HB-HAB) link. Environ Res Lett. 9(10):105001.

24.	Rockström J, Steffen W, Noone K, Scheffer M. 2009. A safe operating space for humanity. Nature. 461:472-475.



## Module 01 references

1. [Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA.95(12):6578–6583.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

2. [Canfield D, Glazer A, Falkowski P. 2010. The Evolution and Future of Earth’s Nitrogen Cycle. Science 330:192-196.](https://www.ncbi.nlm.nih.gov/pubmed/20929768)

3. [Falkowski P, Fenchel T, Delong E. 2008. The Microbial Engines That Drive Earth's Biogeochemical Cycles. Science 320:1034-1039.](https://www.ncbi.nlm.nih.gov/pubmed/18497287)

4. [Kasting JF, Siefert JL. 2002. Life and the Evolution of Earth's Atmosphere. Science. 296:1066-1068.](https://www.ncbi.nlm.nih.gov/pubmed/12004117)

5. [Nisbet E, Sleep N. 2001. The habitat and nature of early life. Nature 409:1083-1091.](https://www.nature.com/articles/35059210)

6. [Falkowski P, Scholes R, Boyle E, Canadell J, Canfield D, Elser J, Gruber N. et al. 2000. The Global Carbon Cycle: A Test of Our Knowledge of Earth as a System. Science 290:291-296.](http://science.sciencemag.org/content/290/5490/291)

7. [Kallmeyer J, Pockalny R, Adhikari R, Smith D, D'Hondt S. 2012. Global distribution of microbial abundance and biomass in subseafloor sediment. Proceedings of the National Academy of Sciences 109:16213-16216.](http://www.pnas.org/content/109/40/16213)

8. [Rockström J, Steffen W, Noone K, Scheffer M. 2009. A safe operating space for humanity. Nature. 461:472-475.](https://www.nature.com/articles/461472a)

9. [Schrag D. 2012. Geobiology of the Anthropocene. In Knoll, A, Canfield, D, Konhauser, K (ed.), Fundamentals of Geobiology. Blackwell Publishing Ltd, New Jersey.](https://onlinelibrary.wiley.com/doi/10.1002/9781118280874.ch22)

10. [Waters C, Zalasiewicz J, Summerhayes C, Barnosky A, Poirier C, Ga uszka A, Cearreta A, Edgeworth M, Ellis E, Ellis M, Jeandel C, Leinfelder R, McNeill J, Richter D, Steffen W, Syvitski J, Vidas D, Wagreich M, Williams M, Zhisheng A, Grinevald J, Odada E, Oreskes N, Wolfe A. 2016. The Anthropocene is functionally and stratigraphically distinct from the Holocene. Science 351:aad2622-aad2622.](http://science.sciencemag.org/content/351/6269/aad2622)

#Module 02

##Evidence worksheets 

###Evidence worksheet 04

[Martinez *et al.* 2007](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1838496/)

####Learning objectives:

* Discuss the relationship between microbial community structure and metabolic diversity

* Evaluate common methods for studying the diversity of microbial communities

* Recognize basic design elements in metagenomic workflows

####Questions:
* What were the main questions being asked?
  1. Can PR system be expressed in *E. coli* to allow for use of light for energy?
  2. What affects wide taxonomic distribution of PR?
  3. What are functions of PR system *in vivo* & how are they different from what was predicted?
  
* What were the primary methodological approaches used?
  1. Functional screening from a fosmid library generated from environmental sample of phytoplankton
  2. HPLC to find expressed photsynthetic pigments
  3. Analysis of proton pumping
  4. Luciferase-based assay to measure ATP

* Summarize the main results or findings. 
  1. 2 clones with genetically distinct PR systems were expressed and identified in *E. coli*
  2. Functions of system components and pathway products were elucidated
  3. Evidence that a single HGT event can confer photosynthetic capabilities, which explains why PR is so widely distributed

* Do new questions arise from the results?
  1. How can we identify functional PR systems in environmental samples?
  2. How much of PR system is exressed in bacteria that encode it?
  3. Are there favourable genetic backgrounds for acquiring PR by HGT?
  4. Why is PR more distributed in marine bacteria?
  5. What activates PR expression?

* Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 
  1. How did they measure the proton pump? Hard to understand.

## Problem sets

###Problem set 03
#### Learning objectives: 
Specific emphasis should be placed on the process used to find the answer. Be as comprehensive as possible e.g. provide URLs for web sources, literature citations, etc.  

#### Specific Questions:
* How many prokaryotic divisions have been described and how many have no cultured representatives (microbial dark matter)?

2016: 89 bacterial and 20 archaeal phyla, from 16S databases
"Shadow biosphere" could contain up to 1500 bacterial phyla

*another team found these numbers from somewhere, but I couldn't find exactly where from*
  

* How many metagenome sequencing projects are currently available in the public domain and what types of environments are they sourced from?

Thousands available on many online metagenome databases, such as [*EBI*](https://www.ebi.ac.uk), which has 110217 projects currently.
Pointless to establish a certain number, because it's always changing.
Sourced from all environments, but esp. where it's hard to culture (gut, ocean, sediment, soil, etc.)
  

* What types of on-line resources are available for warehousing and/or analyzing environmental sequence information (provide names, URLS and applications)?  

1. Shotgun metagenomics:
  EULER (assembly), S-GCOM (binning), KEGG (annotation), Megan 5 (analysis pipelines)
  
2. Marker gene metagenomics:
  OTU base (standalone software), SILVA (analysis pipelines), Amplicon Noise (denoising), RDP (databases)
  
NCBI, MG-RAST

* What is the difference between phylogenetic and functional gene anchors and how can they be used in metagenome analysis?   

    + Phylogenetic:
    Vertical gene transfer, taxonomic, ideally single-copy; Carry phylogenetic information allowing tree reconstruction.
    
    + Functional:
    more HGT, not as useful for phylogeny. Identify specific biochemical functions associated with measurable effects.

* What is metagenomic sequence binning? What types of algorithmic approaches are used to produce sequence bins? What are some risks and opportunities associated with using sequence bins for metabolic reconstruction of uncultivated microorganisms?  

= grouping sequences from a single genome. align sequences to databases --> group together based on DNA characteristics. Unfortunately, prone to contamination from different phylogeny, arbitrary, incomplete genome coverage.

* Is there an alternative to metagenomic shotgun sequencing that can be used to access the metabolic potential of uncultivated microorganisms? What are some risks and opportunities associated with this alternative?  

1. Functional screens (can be hard to express, find host)
2. 3rd gen sequencing (nanopore)
3. Single-cell sequencing
4. FISH probe

## Module 02 references

1. [Madsen E. 2005. Identifying microorganisms responsible for ecologically significant biogeochemical processes. Nature Reviews Microbiology 3:439-446.](https://www.nature.com/articles/nrmicro1151)

2. [Martinez A, Bradley A, Waldbauer J, Summons R, DeLong E. 2007. Proteorhodopsin photosystem gene expression enables photophosphorylation in a heterologous host. Proceedings of the National Academy of Sciences 104:5590-5595.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1838496/)

3. [Taupp M, Mewis K, Hallam S. 2011. The art and design of functional metagenomic screens. Current Opinion in Biotechnology 22:465-472.](https://www.ncbi.nlm.nih.gov/pubmed/21440432)

4. [Wooley J, Godzik A, Friedberg I. 2010. A Primer on Metagenomics. PLoS Computational Biology 6:e1000667.](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000667)

#Module 03

###Evidence worksheets
####Evidence worksheet 05
####Learning objectives

* Evaluate the concept of microbial species based on environmental surveys and cultivation studies.

* Explain the relationship between microdiversity, genomic diversity and metabolic potential

* Comment on the forces mediating divergence and cohesion in natural microbial communities

* Comment on the creative tension between gene loss, duplication and acquisition as it relates to microbial genome evolution

* Identify common molecular signatures used to infer genomic identity and cohesion

* Differentiate between mobile elements and different modes of gene transfer

####Questions
#####Part 1

* What were the main questions being asked?

1. comparison of 3 different strains of E. coli (2 uropathogenic, 1 non-pathogenic)
2.compare genetic similarity/OTU similarity between pathogenic and non-pathogenic strains
3. determine conserved pathogenicity associated genes/PAIs 
4. how lateral gene transfer contributes to the emergence of new uropathogenic E. coli strains

* What were the primary methodological approaches used?

**clones and sequencing:**
Used pBluescript and M13Janus to prepare whole-genome libraries using genomic DNA
Data are collected from Applied Biosystem ABI377 and 3700n automated seqeuncers.
SEQMANII used to assemble sequence data
Primer wlaking, PCR-based analysis used to finish sequencing opposite ends of linking clones
XhoI optical map of whole genome used to order contigs and confirm contig structure, and act as map of whole genome

**sequence analysis and annotation:**
Annotated genome sequence in MAGPIE, defined ORFs in GLIMMER
BLAST used to search for predicted proteins
shotgun sequencing and PCR
Used 90% identity match of CFT073 in either MG1655 or EDL933, 90% of both genes in alignments, and lack of equivalent match in other location in CFT073 genome used to infer orthology


* Summarize the main results or findings.

1. The CFT073 genome is circular, 5 million-bp chromosomal sequence with seven times coverage.
2. Selective pressure to conserve PAI in infection and host colonization
3. The backbone is evolutionarily conserved through vertical gene transfer 
4. Specific regions for insertion of infectious genes are conserved (location is conserved, not specificity to genes)
5. Great genetic diversity between PAIs of pathogenic/pathogenic strains, pathogenic/non-pathogenic
6. Each type/species of E. coli possess combinations of island genes that confer its characteristic lifestyle or its pathogenicity. For example, the uropathogenic strain, CFT073 acquired an  island gene that allows it to infect the urinary tract and bloodstream without compromising its ability to colonize the intestine.
7. net result is a mosaic genome structure in which newly acquired genes in each of the E. coli types are placed into a framework made of genes that distinguishes E. coli from its closer relatives such as S. enterica


* Do new questions arise from the results?

1. How similar/different are pathogenic E. coli strains from other pathogens (ie. Shigella? Y. pestis)?
2. How can species be defined to account for frequent gain and loss of accessory genes cannot simply define species by phenotypic analysis and low resolution mapping alone
3. Why hasnt there been any evidence of extensive genome reductions despite the fact that E.coli has been living commensualy in the intestinal system of animals, which are nutrient rich environments, for millions of years? 


* Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?

Overall gist was clear, however some of the extraneous details in the paper assumed that the reader had a good knowledge of E.coli genomics. 
Paper requires prior understanding of genetics (i.e. the selective pressures which led to the retention of rare codons).

#####Part 2

* Based on your reading and discussion notes, explain the meaning and content of the following figure derived from the comparative genomic analysis of three E. coli genomes by Welch et al. Remember that CFT073 is a uropathogenic strain and that EDL933 is an enterohemorrhagic strain. Explain how this study relates to your understanding of ecotype diversity. Provide a definition of ecotype in the context of the human body. Explain why certain subsets of genes in CFT073 provide adaptive traits under your ecological model and speculate on their mode of vertical descent or gene transfer.

1. Ecotype: varying tissues types (each type of human cell has different receptors and secrete different and surrounding factors that make up the extracellular environment
2. Ecotype in human body -> skin colour depends on the altitude; microbiota in digestive system -> diet;


###Problem sets
####Problem set 04
####Learning objectives

* Gain experience estimating diversity within a hypothetical microbial community

####Questions
####First, I analyzed the data from our total community (large bag)

```{r}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)

#Record data from larger sample
bag_data1 = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51),
  name = c("sp1", "sp2" , "sp3", "sp4", "sp5", "sp6", "sp7", "sp8", "sp9", "sp10",
           "sp11", "sp12", "sp13", "sp14", "sp15", "sp16", "sp17", "sp18", "sp19", "sp20",
           "sp21", "sp22", "sp23", "sp24", "sp25", "sp26", "sp27", "sp28", "sp29", "sp30",
           "sp31", "sp32", "sp33", "sp34", "sp35", "sp36", "sp37", "sp38", "sp39", "sp40",
           "sp41", "sp42", "sp43", "sp44", "sp45", "sp46", "sp47", "sp48", "sp49", "sp50", "sp51"),
  characteristics = c("orange bear", "pink bear", "yellow bear", "white bear", "red bear", "green bear", "pink solid bear", "red solid bear",
                      "yellow bean", "orange bean", "pink bean", "green bean", "red bean",
                      "yellow rod", "orange rod", "red rod", "black rod",
                      "red swirl", "blue swirl",
                      "yellow oval",
                      "red filamentous", 
                      "orange spider", "pink spider", "purple spider",
                      "red bottle", "pink bottle", "yellow bottle",
                      "silver kiss",
                      "blue big brick", "yellow big brick", "pink big brick",
                      "blue small brick", "yellow small brick", "green small brick", "pink small brick",
                      "green sphere", "purple sphere", "red sphere", "orange sphere", "yellow sphere",
                      "orange mnm", "green mnm", "red mnm", "yellow mnm", "brown mnm", "blue mnm",
                      "yellow skittle", "orange skittle", "red skittle", "brown skittle", "green skittle"),
  occurences = c(14,16,16,16,10,18,1,1,26,31,37,33,38,3,1,2,1,1,2,1,5,2,3,1,1,1,1,15,1,1,1,3,3,2,6,5,3,7,5,3,59,30,27,34,30,60,34,39,30,30,42)
)

#Make table
bag_data1 %>% 
  kable("html") %>%
  kable_styling(font_size = 10, full_width = F)

#To calculate alpha-diversity in R
library(vegan)

bag_data1_diversity = 
  bag_data1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

bag_data1_diversity

#Simpson's reciprocal index:
diversity(bag_data1_diversity, index="invsimpson")

#Chao1 richness estimator:
specpool(bag_data1_diversity)
```

####Then, I analyzed data from my subsample (small bag)

```{r}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)

#Recording data from my little bag:
small_data = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51),
  name = c("sp1", "sp2" , "sp3", "sp4", "sp5", "sp6", "sp7", "sp8", "sp9", "sp10",
           "sp11", "sp12", "sp13", "sp14", "sp15", "sp16", "sp17", "sp18", "sp19", "sp20",
           "sp21", "sp22", "sp23", "sp24", "sp25", "sp26", "sp27", "sp28", "sp29", "sp30",
           "sp31", "sp32", "sp33", "sp34", "sp35", "sp36", "sp37", "sp38", "sp39", "sp40",
           "sp41", "sp42", "sp43", "sp44", "sp45", "sp46", "sp47", "sp48", "sp49", "sp50", "sp51"),
  characteristics = c("orange bear", "pink bear", "yellow bear", "white bear", "red bear", "green bear", "pink solid bear", "red solid bear",
                      "yellow bean", "orange bean", "pink bean", "green bean", "red bean",
                      "yellow rod", "orange rod", "red rod", "black rod",
                      "red swirl", "blue swirl",
                      "yellow oval",
                      "red filamentous", 
                      "orange spider", "pink spider", "purple spider",
                      "red bottle", "pink bottle", "yellow bottle",
                      "silver kiss",
                      "blue big brick", "yellow big brick", "pink big brick",
                      "blue small brick", "yellow small brick", "green small brick", "pink small brick",
                      "green sphere", "purple sphere", "red sphere", "orange sphere", "yellow sphere",
                      "orange mnm", "green mnm", "red mnm", "yellow mnm", "brown mnm", "blue mnm",
                      "yellow skittle", "orange skittle", "red skittle", "brown skittle", "green skittle"),
  occurences = c(0,3,1,1,0,1,1,0,8,3,7,9,13,1,1,1,0,0,0,0,0,0,2,0,0,0,0,3,0,0,0,0,0,0,2,1,0,1,0,0,18,6,10,5,3,15,8,12,6,2,13))

#Make table
small_data %>% 
  kable("html") %>%
  kable_styling(font_size = 10, full_width = F)

#To calculate alpha-diversity in R
library(vegan)

small_data_diversity = 
  small_data %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

small_data_diversity

#Simpson's reciprocal index:
diversity(small_data_diversity, index="invsimpson")

#Chao1 richness estimator:
specpool(small_data_diversity)

#Collector's curve for my little bag sample

small_data2 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20),
  y = c(1,2,3,3,4,5,6,7,7,7,8,9,9,10,11,12,13,14,15,16)
)

ggplot(small_data2, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```

####Question answers for all parts

1. Construct a table listing each species, its distinguishing characteristics, the name you have given it, and the number of occurrences of the species in the collection.

Did that, see above

2. Ask yourself if your collection of microbial cells from seawater represents the actual diversity of microorganisms inhabiting waters along the Line-P transect. Were the majority of different species sampled or were many missed?

My sample doesn't look representative enough by eye. Many species were missing and some were too many.

3. For your sample: Create a collector’s curve for your sample (not the entire original community).

Did that, see above.

* Does the curve flatten out? If so, after how many individual cells have been collected?

No, sampled 20 cells

* What can you conclude from the shape of your collector’s curve as to your depth of sampling?

Sampling wasn't sufficient, but my sample wasn't perfect either.

4. What is the Simpson Reciprocal Index for your sample? What is the Simpson Reciprocal Index for your original total community?

Sample: ~16
Community: ~23

5. What is the chao1 estimate for your sample? What is the chao1 estimate for your original total community?

Sample: ~29
Community: ~50

6. What are the Simpson Reciprocal Indices for your sample and community using the R function?

Sample: 15.97
Community: 23.17

I don't think these are very close

7. What are the chao1 estimates for your sample and community using the R function?

Sample: 29
Community: 51

Similar to what I've got.

8. How does the measure of diversity depend on the definition of species in your samples?

We separated candy based on shapes, texture and colour. That way we managed to record "rare" species, such as "pink solid bear". If we only had shape or general shade of colour, we wouldn't be able to distinguish rare from common, and it wouldn't be representative and wouldn't be very diverse.

9. Can you think of alternative ways to cluster or bin your data that might change the observed number of species?

As said previously, we could exclude some parameters we used for classification, like texture, size, shade of colour, etc. We could also add parameters, like exact colour combinations (we recorded two types of spider candy as one, because they both shared half of same colour)

10. How might different sequencing technologies influence observed diversity in a sample?

Sequencing can introduce errors, and based on data analysis methods these could be overlooked to create non-existing species, or instead true different sequences could be grouped together because they are flagged as false alarm, erasing record of one of the two species, for example. This would result in over/underepresentation of species in a community.

###Writing assignment 02

* Prompt: Discuss the challenges involved in defining a microbial species and how HGT complicates matters, especially in the context of the evolution and phylogenetic distribution of microbial metabolic pathways.  Can you comment on how HGT influences the maintenance of global biogeochemical cycles through time?  Finally, do you think it is necessary to have a clear definition of a microbial species?  Why or why not?

**Introduction**

Traditionally, species classification has relied on both genotypic and phenotypic characteristics. However, in large-scale surveys of unculturable organismal communities, characterization with multiple approaches has obvious complications, and use of marker genes, such as the 16S rRNA gene, has been instead adopted for convenience. In order to maintain reliability of classification by this method, nucleotide identity thresholds were also proposed for clustering of sequences into distinct taxa. Despite continuous revision of identity thresholds, it is becoming increasingly apparent that marker genes may be too conserved, and whole genome sequences should rather be used for adequate resolution between closely-related taxa (1). While human and technical biases, such as the lack of consistent prokaryotic species definition and sequencing errors, respectively, can complicate diversity analyses, natural processes, such as the horizontal gene transfer (HGT), further necessitate the use of metagenomics in species classification (1, 2). 

Here, I explore the extent of HGT-mediated contribution to intraspecies diversity, as well as maintenance of global biogeochemical cycles through time. I also discuss some of the challenges associated with prokaryotic species classification, emphasizing the importance of revision of the existing species definition, potentially by using high-throughput approaches.

**Genetic cohesion complicates resolution between closely-related prokaryotes**

HGT is one of the main contributors to genetic cohesion of closely-related microorganisms, mediating high levels of information exchange across nearly all genes, resulting in occurrence of phenotypically distinct taxa that share >97% identity by 16S rRNA methods (3). In addition, organisms co-existing in close communities or occupying same ecological niches also experience significant levels of genetic exchange. As such, Caro-Quintero and Konstantinidis (2015) assessed the extent of HGT across different bacterial phyla using bioinformatics approaches. Their analysis revealed that shared ecology and oxygen tolerance represent the two major driving factors for HGT among distantly-related bacteria, contributing to the total exchange encompassing roughly 35% of genes involved in metabolism, and around 16% of the total genes in some organisms (4).

Such dramatic extent of HGT is also responsible for occurrence of closely-related organisms that display highly diverse characteristics in situ. For example, plant-associated *Pseudomonas* spp. exhibit a range of phenotypes from pathogenic to growth-promoting to immune-modulating, but they are indistinguishable by 16S (5). Similarly, various strains of *Escherichia coli* also display diverse phenotypes and perform distinct ecological roles, while sometimes sharing only 20% of whole genome identity (6, 7).

**HGT conserves metabolic processes through distribution of genetic elements across diverse taxa and ecological niches**

Despite the associated obstacles in prokaryotic classification, HGT remains an exciting phenomenon due to its undeniable contributions to bacterial and archaeal diversity. In addition, HGT mediates maintenance of global biogeochemical cycles through time by establishing interspecies diversity through continuous exchange of metabolic pathways across distant phyla. While environmental selection becomes limiting towards the ultimate survival of a particular genetic element within an organism, HGT ensures survival of associated function on a global scale by distributing genetic information across diverse organisms. As some of these organisms will occupy the desirable ecological niche, they will store the acquired information until it becomes possible to pass the associated process onto another taxon (8). 

This process is reflected by the wide taxonomic distribution of many metabolic pathways in prokaryotes. For example, Tavormina et al. (2010) document presence of enzymes involved in methanotrophy across 3 classes of Proteobacteria (alpha, beta, gamma), as well as several uncharacterized lineages in multiple marine water samples. Such wide distribution of methane oxidation in the ocean is supported by a major contribution of this metabolic process towards breakdown of methane produced in marine sediments, which significantly decreases emission of this greenhouse gas into the atmosphere (9). Similarly, Olson et al. (2018) report presence of proteorhodopsin in metagenomic samples from North Pacific, taken from surface waters and up to 1000 m depth. Further, proteorhodopsin genes were present in diverse bacterial taxa, such as SAR11, SAR86, SUP05, Cyanobacteria and Actinobacteria; as well as in archaea (10). Altogether, the study by Olson et al. provides undeniable evidence of high ecological relevance of proteorhodopsin-driven photosynthesis in marine communities, while also suggesting the HGT-driven origin for the observed distribution of associated genes.

**Need for a revised species definition to be resolved with high-throughput approaches**

The definition of species is an essential element of any study investigating organismal communities, as it delineates the core unit of biodiversity. However, there is no general consensus regarding the exact definition of species in the context of prokaryotic life, and scientists, as a consequence, often stretch its boundaries in agreement with the demands and objectives dictated by the particular nature of their study. This leads to inconsistencies and instability in taxonomic classification, which does not only inconvenience researchers, but, ultimately, complicates future  particularly, interdisciplinary  studies (11). For this reason, a coherent definition of species is absolutely required across all forms of life. 

With current technologies, multiple approaches have become available to facilitate the correction using measurable phenotypic, genotypic, ecological and evolutionary parameters. As such, metagenomic surveys confirm presence of genetically and ecologically discrete populations, rather than a genetic and ecological continuum, in natural habitats, allowing for in-depth resolution between closely-related taxa (3, 12, 13). Additional studies also suggest use of culture-independent techniques in combination with experimental approaches for more efficient classification, based on evolutionary and functional cues (14). Although the latter may not be achievable with yet-uncultivable organisms, Konstantinidis et al. (2017) argue that metagenomics alone can still provide resolution at sufficiently acceptable levels (15). Information about members of particular taxa, gathered using high-throughput approaches, will assist in delineating core characteristics of a microbial species.

**Conclusion**

Altogether, HGT serves as a major contributor to conservation of global biogeochemical processes and the associated metabolic pathways in prokaryotes. This genetic exchange also mediates emergence of intraspecies diversity, creating many challenges in classification of organisms with the aim to infer their ecological functions within natural habitats. Despite the influence of HGT, many of the obstacles faced by researchers today can be overcome by revising the current prokaryotic species definition using high-throughput approaches, such as metagenomics. However, in order to produce a coherent, universal species definition, the proposed approaches must be applied with great consistency. With this, the revised definition becomes an achievable goal only through multi-disciplinary cooperation.

**References**

1. Rodriguez-R L, Castro J, Kyrpides N, Cole J, Tiedje J, Konstantinidis K. 2018. How Much Do rRNA Gene Surveys Underestimate Extant Bacterial Diversity? Appl Environ Microbiol 84:e00014-18.

2. Koonin E, Makarova K, Aravind L. 2001. Horizontal Gene Transfer in Prokaryotes: Quantification and Classification. Annu Rev Microbiol 55:709-742.

3. Caro-Quintero A, Konstantinidis K. 2011. Bacterial species may exist, metagenomics reveal. Environ Microbiol 14:347-355.

4. Caro-Quintero A, Konstantinidis K. 2014. Inter-phylum HGT has shaped the metabolism of many mesophilic and anaerobic bacteria. ISME J 9:958-967.

5. Haney C, Wiesmann C, Shapiro L, Melnyk R, O'Sullivan L, Khorasani S, Xiao L, Han J, Bush J, Carrillo J, Pierce N, Ausubel F. 2017. Rhizosphere-associated Pseudomonas induce systemic resistance to herbivores at the cost of susceptibility to bacterial pathogens. Mol Ecol.

6. Anderson M, Whitlock J, Harwood V. 2006. Diversity and Distribution of Escherichia coli Genotypes and Antibiotic Resistance Phenotypes in Feces of Humans, Cattle, and Horses. Appl Environ Microbiol 72:6914-6922.

7. Lukjancenko O, Wassenaar T, Ussery D. 2010. Comparison of 61 Sequenced Escherichia coli Genomes. Microb Ecol 60:708-720.

8. Falkowski P, Fenchel T, Delong E. 2008. The Microbial Engines That Drive Earth's Biogeochemical Cycles. Science 320:1034-1039.

9. Tavormina P, Ussler W, Joye S, Harrison B, Orphan V. 2010. Distributions of putative aerobic methanotrophs in diverse pelagic marine environments. ISME J 4:700-710.

10. Olson D, Yoshizawa S, Boeuf D, Iwasaki W, DeLong E. 2018. Proteorhodopsin variability and distribution in the North Pacific Subtropical Gyre. ISME J 12:1047-1060.

11. Avise J, Liu J. 2011. On the temporal inconsistencies of Linnean taxonomic ranks. Biol J Linn Soc 102:707-714.

12. Bowman J, Ducklow H. 2015. Microbial Communities Can Be Described by Metabolic Structure: A General Framework and Application to a Seasonally Variable, Depth-Stratified Microbial Community from the Coastal West Antarctic Peninsula. PLoS One 10:e0135868.

13. Palmer M, Steenkamp E, Coetzee M, Blom J, Venter S. 2018. Genome-Based Characterization of Biological Processes That Differentiate Closely Related Bacteria. Front Microbiol 9:113.

14. Ellegaard K, Engel P. 2016. Beyond 16S rRNA Community Profiling: Intra-Species Diversity in the Gut Microbiota. Front Microbiol 7:1475.

15. Konstantinidis K, Rossello-Mora R, Amann R. 2017. Uncultivated microbes in need of their own taxonomy. ISME J 11:2399-2406.

###Module 03 references

1. [Callahan B, McMurdie P, Holmes S. 2017. Exact sequence variants should replace operational taxonomic units in marker-gene data analysis. The ISME Journal 11:2639-2643.](https://www.ncbi.nlm.nih.gov/pubmed/28731476)

2. [Gaudet A, Ramer L, Nakonechny J, Cragg J, Ramer M. 2010. Small-Group Learning in an Upper-Level University Biology Class Enhances Academic Performance and Student Attitudes Toward Group Work. PLoS ONE 5:e15821.](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0015821)

3. [Hallam S, Torres-Beltrán M, Hawley A. 2017. Monitoring microbial responses to ocean deoxygenation in a model oxygen minimum zone. Scientific Data 4:170158.](https://www.nature.com/articles/sdata2017158)

4. [Hawley A, Torres-Beltrán M, Zaikova E, Walsh D, Mueller A, Scofield M, Kheirandish S, Payne C, Pakhomova L, Bhatia M, Shevchuk O, Gies E, Fairley D, Malfatti S, Norbeck A, Brewer H, Pasa-Tolic L, del Rio T, Suttle C, Tringe S, Hallam S. 2017. A compendium of multi-omic sequence information from the Saanich Inlet water column. Scientific Data 4:170160.](https://www.nature.com/articles/sdata2017160)

5. [Kunin V, Engelbrektson A, Ochman H, Hugenholtz P. 2010. Wrinkles in the rare biosphere: pyrosequencing errors can lead to artificial inflation of diversity estimates. Environmental Microbiology 12:118-123.](https://www.ncbi.nlm.nih.gov/pubmed/19725865)

6. [Cordero O, Ventouras L, DeLong E, Polz M. 2012. Public good dynamics drive evolution of iron acquisition strategies in natural bacterioplankton populations. Proceedings of the National Academy of Sciences 109:20059-20064.](https://www.ncbi.nlm.nih.gov/pubmed/23169633)

7. [Giovannoni S. 2012. Vitamins in the sea. Proceedings of the National Academy of Sciences 109:13888-13889.](http://www.pnas.org/content/109/35/13888)

8. [Lundin D, Severin I, Logue J, Östman Ö, Andersson A, Lindström E. 2012. Which sequencing depth is sufficient to describe patterns in bacterial α- and β-diversity?. Environmental Microbiology Reports 4:367-372.](https://www.ncbi.nlm.nih.gov/pubmed/23760801)

9. [Morris J, Lenski R, Zinser E. 2012. The Black Queen Hypothesis: Evolution of Dependencies through Adaptive Gene Loss. mBio 3:e00036-12-e00036-12.](https://www.ncbi.nlm.nih.gov/pubmed/22448042)

10. [Thompson J. et al. 2005. Genotypic Diversity Within a Natural Coastal Bacterioplankton Population. Science 307:1311-1313.](https://www.ncbi.nlm.nih.gov/pubmed/15731455)

11. [Sogin M, Morrison H, Huber J, Welch D, Huse S, Neal P, Arrieta J, Herndl G. 2006. Microbial diversity in the deep sea and the underexplored "rare biosphere". Proceedings of the National Academy of Sciences 103:12115-12120.](http://www.pnas.org/content/103/32/12115)

12. [Torres-Beltrán M, Hawley A, Capelle D, Zaikova E, Walsh D, Mueller A, Scofield M, Payne C, Pakhomova L, Kheirandish S, Finke J, Bhatia M, Shevchuk O, Gies E, Fairley D, Michiels C, Suttle C, Whitney F, Crowe S, Tortell P, Hallam S. 2017. A compendium of geochemical information from the Saanich Inlet water column. Scientific Data 4:170159.](https://www.nature.com/articles/sdata2017159)

13. [Welch R, Burland V, Plunkett G, Redford P, Roesch P, Rasko D, Buckles E, Liou S, Boutin A, Hackett J, Stroud D, Mayhew G, Rose D, Zhou S, Schwartz D, Perna N, Mobley H, Donnenberg M, Blattner F. 2002. Extensive mosaic structure revealed by the complete genome sequence of uropathogenic Escherichia coli. Proceedings of the National Academy of Sciences 99:17020-17024.](https://www.ncbi.nlm.nih.gov/pubmed/12471157)

##Projects
###Project 1 (Group 7)

**Comments:**

I was having trouble with my laptop functioning properly knitting such a big file together. 
Because of this, I attached a link to separate html files containing each project.

[Here is the Project 1 html document](https://drive.google.com/file/d/10iT0ezvgqjUQ75x0XL_17nKuiTrQOFQI/view?usp=sharing)

###Project 2 (Group 7)

**Comments:**

I was having trouble with my laptop functioning properly knitting such a big file together. 
Because of this, I attached a link to separate html files containing each project.

[Here is the Project 2 html document](https://drive.google.com/open?id=1dmVhXdna23K0E73QJ3KB3P8F-ZD59PYh)